CSP/CspInstance.cs

```diff
@@ -72,8 +87,34 @@ public void RemoveColor(Pair pair)
             pair.Variable.avalibleColors.Remove(pair.Color);
         }
         public void RemoveColor(Variable variable, Color color) => RemoveColor(new Pair(variable, color));
-       
 
+        #endregion
+
+        #region variable
+        public void AddVariable(IEnumerable<Color> variableColors)
```

Comment:
Jeśli dodałeś funkcje do dodawania i usuwania zmiennych to może warto zmienić Variables z ISet na IReadonlySet? Będzie trzeba poprawić to w moich lematach, ale nie powinno być tego dużo